
<launch>
  <env name="ROSCONSOLE_FORMAT" value="[${severity}][${thread}][${node}/${function}:${line}]: ${message}"/>
  <env name="ROS_LOG_DIR" value="/tmp/"/>

  <arg name="map_frame" default="map" />
  <arg name="base_frame" default="base_stabilized"/>
  <arg name="odom_frame" default="base_frame"/>

  <arg name="scan_topic" default="scan"/>

  <arg name="use_external_imu" type="bool" default="false"/>
  <arg name="external_imu_topic" default=""/>

  <node pkg="robot_platform" type="robot_platform" name="robot_platform" output="screen" respawn="true">

    <param name="serial_dev" type="string" value="/dev/serial/by-id/usb-Teensyduino_USB_Serial_13671670-if00" />
    <param name="serial_baudrate" type="int" value="115200" />

    <param name="raw_input_topic" type="string" value="/robot_platform/raw_serial_request" />
    <param name="goal_move_input_topic" type="string" value="/move_base_simple/goal" />
  
    <param name="use_external_imu" type="bool" value="$(arg use_external_imu)" />
    <param name="external_imu_topic" type="string" value="$(arg external_imu_topic)" />
    
    <param name="raw_output_topic" type="string" value="/robot_platform/raw_serial_response" />
    <param name="battery_state_output_topic" type="string" value="/robot_platform/battery" />
    <param name="gps_state_output_topic" type="string" value="/robot_platform/gps" />
    <param name="imu_state_output_topic" type="string" value="/robot_platform/imu" />

    <param name="map_frame_id" type="string" value="$(arg map_frame)" />
    <param name="robot_to_map_projection_frame_id" type="string" value="$(arg base_frame)" />
    <param name="robot_frame_id" type="string" value="$(arg odom_frame)" />
    <param name="rplidar_frame_id" type="string" value="$(arg scan_topic)" />
  </node>

</launch>
